File: SecureOnlineVotingSystem/SecureVoting.API/Models/VoterVerification.cs
namespace SecureVoting.API.Models
{
    public class VoterVerification
    {
        public int VerificationId { get; set; }
        public int UserId { get; set; }

        public string LegalFullName { get; set; } = "";
        public DateTime? DateOfBirth { get; set; }

        public string AddressLine1 { get; set; } = "";
        public string? AddressLine2 { get; set; }
        public string City { get; set; } = "";
        public string StateCode { get; set; } = "";
        public string ZipCode { get; set; } = "";

        public int? JurisdictionId { get; set; }

        public string IdDocumentType { get; set; } = "";
        public string? IdDocumentNumberMasked { get; set; }
        public string? IdDocumentState { get; set; }

        public bool IsIdentityVerified { get; set; }
        public bool IsResidenceVerified { get; set; }
        public bool IsEligibleToVote { get; set; }

        public string VerificationStatus { get; set; } = "";
        public string? ReviewerNotes { get; set; }
        public int? ReviewedByUserId { get; set; }
        public DateTime? ReviewedAtUtc { get; set; }

        public DateTime SubmittedAtUtc { get; set; }
        public DateTime UpdatedAtUtc { get; set; }
        public string? PhoneNumber { get; set; }

        public string? IdPictureFileName { get; set; }
        public string? IdPicturePath { get; set; }
    }
}

File: SecureOnlineVotingSystem/securevoting-web/src/app/services/voter-verification.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { environment } from '../../environments/environment';

export interface VoterVerificationPayload {
  legalFullName: string;
  dateOfBirth?: string | null;
  addressLine1: string;
  addressLine2?: string | null;
  city: string;
  stateCode: string;
  zipCode: string;
  jurisdictionId?: number | null;
  idDocumentType: string;
  idDocumentNumberMasked?: string | null;
  idDocumentState?: string | null;
}

@Injectable({
  providedIn: 'root'
})
export class VoterVerificationService {
  private base = environment.apiUrl;

  constructor(private http: HttpClient) {}

  submit(payload: VoterVerificationPayload) {
    return this.http.post<any>(`${this.base}/api/voter-verification/submit`, payload);
  }

  getMine() {
    return this.http.get<any>(`${this.base}/api/voter-verification/me`);
  }

  getPending() {
    return this.http.get<any[]>(`${this.base}/api/voter-verification/pending`);
  }

  getAll() {
    return this.http.get<any[]>(`${this.base}/api/voter-verification/all`);
  }

  getJurisdictions() {
    return this.http.get<any[]>(`${this.base}/api/jurisdictions`);
  }

  approve(userId: number, reviewerNotes: string, jurisdictionId: number) {
    return this.http.post<any>(`${this.base}/api/voter-verification/${userId}/approve`, {
      reviewerNotes: reviewerNotes || null,
      jurisdictionId
    });
  }

  reject(userId: number, reviewerNotes: string) {
    return this.http.post<any>(`${this.base}/api/voter-verification/${userId}/reject`, {
      reviewerNotes
    });
  }
}